pkg/splunk/enterprise/util.go

```diff
@@ -211,6 +217,85 @@ func GetSmartstoreRemoteVolumeSecrets(volume enterprisev1.VolumeSpec, client spl
 	return accessKey, secretKey, namespaceScopedSecret.ResourceVersion, nil
 }
 
+// ApplyAppListingConfigMap creates the configMap  with two entries:
+// (1) app-list-local.yaml
+// (2) app-list-cluster.yaml
+func ApplyAppListingConfigMap(client splcommon.ControllerClient, cr splcommon.MetaObject,
+	appConf *enterprisev1.AppFrameworkSpec, appsSrcDeployStatus map[string]enterprisev1.AppSrcDeployInfo) (*corev1.ConfigMap, bool, error) {
+
+	var err error
+	var crKind string
+	var configMapDataChanged bool
+	crKind = cr.GetObjectKind().GroupVersionKind().Kind
+
+	scopedLog := log.WithName("ApplyAppListingConfigMap").WithValues("kind", crKind, "name", cr.GetName(), "namespace", cr.GetNamespace())
+
+	mapAppListing := make(map[string]string)
+
+	// ToDo: sgontla/Jeff: Adapt to any Ansible changes(if required) for differentiating local vs. cluster scope (?)
+	yamlConfHeader := fmt.Sprintf(`splunk:
+  apps_location:`)
+
+	var localAppsConf, clusterAppsConf string
+	localAppsConf = yamlConfHeader
+	clusterAppsConf = yamlConfHeader
+
+	for appSrc, appSrcDeployInfo := range appsSrcDeployStatus {
+		appDeployList := appSrcDeployInfo.AppDeploymentInfoList
+
+		switch scope := getAppSrcScope(appConf, appSrc); scope {
+		case "local":
+			for idx := range appDeployList {
+				if appDeployList[idx].DeployStatus == enterprisev1.DeployStatusPending {
+					localAppsConf = fmt.Sprintf(`%s
+    - "/init-apps/%s/%s"`, localAppsConf, appSrc, appDeployList[idx].AppName)
+				}
+			}
+
+		case "cluster":
+			for idx := range appDeployList {
+				if appDeployList[idx].DeployStatus == enterprisev1.DeployStatusPending {
+					clusterAppsConf = fmt.Sprintf(`%s
+    - "/init-apps/%s/%s"`, clusterAppsConf, appSrc, appDeployList[idx].AppName)
+				}
+			}
```

Comment:
From the initial look, it seems like the content of both `localAppsConf` and `clusterAppsConf` is of a ### similar style and has nothing to do with scope, right? All they contain is something like - 
``` 
splunk:
  apps_location:
      - "/init-apps/admin_app/admin_app.tgz"
```
If this is the case, then probably make the above thing in a function and pass clusterAppsConf/localAppsConf as a arg to that function?

Reply:
Ya, we can do that. Will address this as part of addressing Jeff's ansible changes.

Reply:
This is a current limitation of the ansible app installation process.  The `apps_location` parameter installs apps locally on non-cluster instances (Standalone/LM) and to the cluster for CM/Deployer.  Once changes are in ansible to add the local scope to CM/Deployer, we'll have to change the `scope:local` case for CM/Deployer here to use the new local install option.